src/sql-parser/src/parser.rs

```diff
@@ -5188,18 +5191,32 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_copy_option(&mut self) -> Result<CopyOption<Raw>, ParserError> {
-        let name =
-            match self.expect_one_of_keywords(&[FORMAT, DELIMITER, NULL, ESCAPE, QUOTE, HEADER])? {
-                FORMAT => CopyOptionName::Format,
-                DELIMITER => CopyOptionName::Delimiter,
-                NULL => CopyOptionName::Null,
-                ESCAPE => CopyOptionName::Escape,
-                QUOTE => CopyOptionName::Quote,
-                HEADER => CopyOptionName::Header,
-                _ => unreachable!(),
-            };
-        let value = self.parse_optional_option_value()?;
-        Ok(CopyOption { name, value })
+        let name = match self
+            .expect_one_of_keywords(&[FORMAT, DELIMITER, NULL, ESCAPE, QUOTE, HEADER, AWS, MAX])?
+        {
+            FORMAT => CopyOptionName::Format,
+            DELIMITER => CopyOptionName::Delimiter,
+            NULL => CopyOptionName::Null,
+            ESCAPE => CopyOptionName::Escape,
+            QUOTE => CopyOptionName::Quote,
+            HEADER => CopyOptionName::Header,
+            AWS => {
+                self.expect_keyword(CONNECTION)?;
+                return Ok(CopyOption {
+                    name: CopyOptionName::AwsConnection,
+                    value: Some(self.parse_object_option_value()?),
```

Comment:
Here, `parse_object_option_value` actually directly gives me the resolved `GlobalId` for the connection object which reduces an extra hop because I had to resolve the name and then get the item with `UnresolvedItemName` (and this works, I tested this out).

I saw this getting used in a similar fashion here https://github.com/MaterializeInc/materialize/blob/main/src/sql-parser/src/parser.rs#L2487-L2491, where it's getting an AwsPrivateLink connection.

Reply:
Btw, I think the other connection options should actually use this instead as well? Since this is fully resolved?

Reply:
Good find! Yeah, seems like the others are just wrong then.